File: lib/card.cc
#include "card.h"

#include <string>

bool Card::operator==(const Card& other) {
  return value == other.value && suit == other.suit;
}

std::string GetNonJokerValueString(const Value value) {
  switch (value) {
    case Value::kThree:
      return "Three";
    case Value::kFour:
      return "Four";
    case Value::kFive:
      return "Five";
    case Value::kSix:
      return "Six";
    case Value::kSeven:
      return "Seven";
    case Value::kEight:
      return "Eight";
    case Value::kNine:
      return "Nine";
    case Value::kTen:
      return "Ten";
    case Value::kJack:
      return "Jack";
    case Value::kQueen:
      return "Queen";
    case Value::kKing:
      return "King";
    case Value::kAce:
      return "Ace";
    case Value::kTwo:
      return "Two";
    default:
      return "";
  }
}

std::string GetSuitString(const Suit suit) {
  switch (suit) {
    case Suit::kClubs:
      return "clubs";
    case Suit::kDiamonds:
      return "diamonds";
    case Suit::kHearts:
      return "hearts";
    case Suit::kSpades:
      return "spades";
    default:
      return "";
  }
}

std::ostream& operator<<(std::ostream& os, const Card& card) {
  switch (card.value) {
    case Value::kBlackJoker:
      os << "Black joker";
      break;
    case Value::kRedJoker:
      os << "Red joker";
      break;
    default:
      os << GetNonJokerValueString(card.value)
         << " of "
         << GetSuitString(card.suit);
  }
  return os;
}

bool CompareCards(const Card card, const Card other_card) {
	if (card.value == other_card.value) {
    return card.suit < other_card.suit;
  }
  return card.value < other_card.value;
}

File: lib/card.py
from enum import Enum

class Value(Enum):
  THREE = 1
  FOUR = 2
  FIVE = 3
  SIX = 4
  SEVEN = 5
  EIGHT = 6
  NINE = 7
  TEN = 8
  JACK = 9
  QUEEN = 10
  KING = 11
  ACE = 12
  TWO = 13
  BLACK_JOKER = 14
  RED_JOKER = 15

class Suit(Enum):
  CLUBS = 1
  DIAMONDS = 2
  HEARTS = 3
  SPADES  = 4

class Card:
  def __init__(self, value, suit = None, is_starting_three_of_clubs = False):
    self.value = value
    self.suit = suit
    self.is_starting_three_of_clubs = is_starting_three_of_clubs

  def __eq__(self, other):
    if not isinstance(other, Card):
      return NotImplemented
    return self.value == other.value && self.suit == other.suit && self.is_starting_three_of_clubs == other.is_starting_three_of_clubs
